packages/web/src/session/session.ts

```diff
@@ -179,16 +174,15 @@ export function initSessionTracking(
 	return {
 		deinit: () => {
 			ACTIVITY_EVENTS.forEach((type) => document.removeEventListener(type, markActivity))
-			rumSessionId = undefined
 			eventTarget = undefined
 		},
 	}
 }
 
-export function getRumSessionId(): SessionId | undefined {
+export function getRumSessionId({ useLocalStorage }: { useLocalStorage: boolean }): SessionId | undefined {
 	if (hasNativeSessionId()) {
 		return window['SplunkRumNative'].getNativeSessionId()
 	}
 
-	return rumSessionId
+	return getCurrentSessionState({ useLocalStorage, forceStoreRead: true })?.id
```

Comment:
When calling `getRumSessionId` we always read the fresh value from storage. Explained in the PR descriptions.